tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java

```diff
@@ -2620,7 +2622,44 @@ else if(questionStatus == Status.FAILED)
 							UIOutput.make(tableRow, "questionShowPoll", String.valueOf(i.getAttribute("questionShowPoll")));
 						}
 					}
-				}  else {
+				} else if(i.getType() == SimplePageItem.TWITTER) {
+					UIOutput.make(tableRow, "twitterSpan");
+					if (canSeeAll) {
+						String itemGroupString = simplePageBean.getItemGroupString(i, null, true);
+						String itemGroupTitles = simplePageBean.getItemGroupTitles(itemGroupString, i);
+						if (itemGroupTitles != null) {
+							itemGroupTitles = "[" + itemGroupTitles + "]";
+						}
+						UIOutput.make(tableRow, "item-groups-titles-text", itemGroupTitles);
+					}
+					if(canSeeAll || simplePageBean.isItemAvailable(i)) {
+						UIVerbatim.make(tableRow, "content", (i.getHtml() == null ? "" : i.getHtml()));
+						//Extract tweet number variable from the html
+						String[] stringArray = i.getHtml().split("data-tweet-limit=");
+						//splitting the remaining text using space as delimiter
+						String[] twitterVariables = stringArray[1].split("\\s+");
+						String tweetLimit = twitterVariables[0];
+						String height = twitterVariables[2].split("=")[1];
+						String[] usernameArray = twitterVariables[3].split(">");
```

Comment:
This needs to be a little safer, as if the HTML isn't exactly as you expect you'll get the `NullPointerException` and I think the page will become uneditable.